WordPress/WordPressTest/EUUSCompliance/CompliancePopoverViewModelTests.swift

```diff
@@ -0,0 +1,87 @@
+import XCTest
+@testable import WordPress
+
+final class CompliancePopoverViewModelTests: CoreDataTestCase {
+    let testDefaults = UserDefaults(suiteName: "compliance-popover-view-model-tests")
+
+    override func setUp() {
+        super.setUp()
+        testDefaults?.removeObject(forKey: UserDefaults.didShowCompliancePopupKey)
+
+        let windowManager = WindowManager(window: UIWindow())
+        WordPressAuthenticationManager(
+            windowManager: windowManager,
+            remoteFeaturesStore: RemoteFeatureFlagStore()
+        ).initializeWordPressAuthenticator()
+    }
+
+    override func tearDown() {
+        super.tearDown()
+        testDefaults?.removeObject(forKey: UserDefaults.didShowCompliancePopupKey)
+    }
+
+    func testDidTapSettingsUpdatesDefaults() {
+        guard let defaults = try? XCTUnwrap(testDefaults) else {
+            return
+        }
+        let sut = CompliancePopoverViewModel(defaults: defaults, contextManager: contextManager)
+        sut.didTapSettings()
+        XCTAssert(defaults.didShowCompliancePopup)
+    }
```

Comment:
```suggestion
    func testDidTapSettingsUpdatesDefaults() throws {
        let defaults = try XCTUnwrap(testDefaults)
        let sut = CompliancePopoverViewModel(defaults: defaults, contextManager: contextManager)
        sut.didTapSettings()
        XCTAssert(defaults.didShowCompliancePopup)
    }
```

Exiting the test early if your assumptions are invalid is dangerous – it risks your tests passing when they're broken.

The pattern above is preferable for error handling within tests.

Reply:
I understand your concern, and I can apply those changes. But `let defaults = try? XCTUnwrap(testDefaults)` can only fail if creating the testDefaults with suiteName fails. Which can only happen if somebody goes and changes the string of the suiteName to a string that cannot be used as a file name. I wouldn't really consider this as a blocker for the PR. One could even force unwrap something like `URL(string: https://wordpress.com)!` as it can only fail if someone deliberately breaks that specific line OR Apple breaks the URL intialization.